src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,213 @@
+        String query = "INSERT INTO cars (manufacturer_id, model) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createCarStatement = connection.prepareStatement(query,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            createCarStatement.setLong(1, car.getManufacturer().getId());
+            createCarStatement.setString(2, car.getModel());
+            createCarStatement.executeUpdate();
+            ResultSet generatedKeys = createCarStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                car.setId(generatedKeys.getObject(1, Long.class));
+            }
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create "
+                    + car + ". ", e);
+        }
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String query = "SELECT c.id AS car_id, manufacturers.id AS manufacturer_id,"
+                + " manufacturers.name AS manufacturer_name, manufacturers.country"
+                + " AS manufacturer_country, model FROM cars c"
+                + " JOIN manufacturers  ON c.manufacturer_id = manufacturers.id"
+                + " WHERE c.id = ? AND c.is_deleted = FALSE;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getCarStatement = connection.prepareStatement(query,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            getCarStatement.setLong(1, id);
+            ResultSet resultSet = getCarStatement.executeQuery();
+            if (resultSet.next()) {
+                car = getCar(resultSet);
+            }
+            assert car != null;
+            car.setDrivers(getCarDrivers(id));
+            return Optional.of(car);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get car by id " + id + ". ", e);
+        }
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT c.id AS car_id, manufacturers.id AS manufacturer_id,"
+                + " manufacturers.name AS manufacturer_name,"
+                + " manufacturers.country AS manufacturer_country, model "
+                + "FROM cars c JOIN manufacturers  ON c.manufacturer_id = manufacturers.id "
+                + "WHERE c.is_deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllCarsStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = getAllCarsStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get a list of cars from carsDB.", e);
+        }
```

Comment:
There's you didn't included car drivers in cars. I think you should add drivers to cars in foreach after cars list is filled and connection is closed

Reply:
I see you calling getCarDrivers in getCar method, which is not good because you create new connection to db while old one still opened